newlib/libc/include/machine/endian.h

```diff
@@ -54,7 +54,7 @@ __bswap64(__uint64_t _x)
 #endif /* !__GNUC__ */
 
 /* endian(3) - similar to linux <endian.h> */
-#if __BYTE_ORDER == __LITTLE_ENDIAN
+#if _BYTE_ORDER == _LITTLE_ENDIAN
```

Comment:
Why not keep both forms? That is,
```c
#if (defined(_BYTE_ORDER) && (_BYTE_ORDER == _LITTLE_ENDIAN)) || (defined(__BYTE_ORDER) && (__BYTE_ORDER == __LITTLE_ENDIAN))
```

Reply:
seems like a good plan to me too; will make sure we don't break existing uses. Should we be checking these with `#ifndef\n#error` to make sure we don't get the wrong value?

Reply:
I went through it all. The double underscores was a regression in my own commit a while ago. The only reason it worked for me was that both little endian and byte order with double underscores was undefined. And undefined equals undefined.

Reply:
> I went through it all. The double underscores was a regression in my own commit a while ago. The only reason it worked for me was that both little endian and byte order with double underscores was undefined. And undefined equals undefined.

Thanks for the explanation. 264a23fb3dafc31a4fa8171cf52dc3877cc02cd8 contains the original mistake in using double underscore, this is just fixing that. Sorry for not catching the error when we merged it!